src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -47,67 +45,56 @@ import {
 import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
 import { useAppSelector } from 'state/store'
 
-import { getSwapperManager } from '../hooks/useSwapper/swapperManager'
 import type { TS } from '../types'
 import { TradeRoutePaths } from '../types'
 import { WithBackButton } from '../WithBackButton'
 import { AssetToAsset } from './AssetToAsset'
 import { ReceiveSummary } from './ReceiveSummary'
 
-type TradeConfirmParams = {
-  fiatRate: string
-}
-
 export const TradeConfirm = ({ history }: RouterProps) => {
   const borderColor = useColorModeValue('gray.100', 'gray.750')
   const [sellTxid, setSellTxid] = useState('')
   const [buyTxid, setBuyTxid] = useState('')
-  const flags = useSelector(selectFeatureFlags)
-  const [swapperName, setSwapperName] = useState<string>('')
   const [executedTradeAmountConstants, setExecutedTradeAmountConstants] =
     useState<ReturnType<typeof getTradeAmountConstants>>()
+  const { bestTradeSwapper } = useSwapper()
   const {
-    getValues,
     handleSubmit,
+    setValue,
+    control,
     formState: { isSubmitting },
   } = useFormContext<TS>()
   const translate = useTranslate()
   const osmosisAsset = useAppSelector(state => selectAssetById(state, osmosisAssetId))
-  const {
-    trade,
-    fees,
-    sellAssetFiatRate,
-    slippage,
-    buyAssetAccountId,
-    sellAssetAccountId,
-    buyTradeAsset,
-  }: Pick<
-    TS,
-    | 'sellAssetAccountId'
-    | 'buyAssetAccountId'
-    | 'trade'
-    | 'fees'
-    | 'sellAssetFiatRate'
-    | 'slippage'
-    | 'buyTradeAsset'
-  > = getValues()
-  const { executeQuote, reset, getTradeTxs } = useSwapper()
+
+  const trade = useWatch({ control, name: 'trade' })
+  const fees = useWatch({ control, name: 'fees' })
+  const sellAssetFiatRate = useWatch({ control, name: 'sellAssetFiatRate' })
+  const feeAssetFiatRate = useWatch({ control, name: 'feeAssetFiatRate' })
+  const slippage = useWatch({ control, name: 'slippage' })
+  const buyAssetAccountId = useWatch({ control, name: 'buyAssetAccountId' })
+  const sellAssetAccountId = useWatch({ control, name: 'sellAssetAccountId' })
+  const buyTradeAsset = useWatch({ control, name: 'buyTradeAsset' })
```

Comment:
Ditto, nice one, for anyone reviewing, separate `useWatch()` statements bring better types